x/finality/keeper/gov.go

```diff
@@ -151,14 +152,8 @@ func (k Keeper) HandleResumeFinalityProposal(ctx sdk.Context, fpPksHex []string,
 		k.SetVotingPowerDistCache(ctx, blkHeight, newDc)
 
 		// ensure every active finality provider has signing info
-		// TODO: check if we should emit events at every height
-		// call HandleFPStateUpdates only at the current height
-		// to account for the last changes (jailed FPs on the proposal)
-		if isCurrHeight {
-			k.HandleFPStateUpdates(ctx, dc, newDc, state)
-			continue
-		}
-		k.HandleFPStateUpdates(ctx, dc, newDc, nil)
+		// Calls empty state until reaches the current block height
+		k.HandleFPStateUpdates(ctx, state, dc, newDc)
```

Comment:
do we need to emit events at every height or should we only call the full HandleFPStateUpdates logic for the current height?
If we should only call this for the current height, we can remove the HandleFPStateUpdates call outside the loop (after it)
And with the current changes I believe we should do that, because otherwise we may be activating same FP several times and calling the hook every time in the `processActiveFp` func

Reply:
modified at https://github.com/babylonlabs-io/babylon/pull/1713/commits/0254c176718cea1bb16ecfe7d937546b0d9c9767